src/chain/mod.rs

```diff
@@ -304,73 +304,88 @@ impl ChainSource {
 		channel_manager: Arc<ChannelManager>, chain_monitor: Arc<ChainMonitor>,
 		output_sweeper: Arc<Sweeper>,
 	) {
-		match self {
-			Self::Esplora { sync_config, logger, .. } => {
-				// Setup syncing intervals if enabled
-				if let Some(background_sync_config) = sync_config.background_sync_config {
-					let onchain_wallet_sync_interval_secs = background_sync_config
-						.onchain_wallet_sync_interval_secs
-						.max(WALLET_SYNC_INTERVAL_MINIMUM_SECS);
-					let mut onchain_wallet_sync_interval = tokio::time::interval(
-						Duration::from_secs(onchain_wallet_sync_interval_secs),
-					);
-					onchain_wallet_sync_interval
-						.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
-
-					let fee_rate_cache_update_interval_secs = background_sync_config
-						.fee_rate_cache_update_interval_secs
-						.max(WALLET_SYNC_INTERVAL_MINIMUM_SECS);
-					let mut fee_rate_update_interval = tokio::time::interval(Duration::from_secs(
-						fee_rate_cache_update_interval_secs,
-					));
-					// When starting up, we just blocked on updating, so skip the first tick.
-					fee_rate_update_interval.reset();
-					fee_rate_update_interval
-						.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
-
-					let lightning_wallet_sync_interval_secs = background_sync_config
-						.lightning_wallet_sync_interval_secs
-						.max(WALLET_SYNC_INTERVAL_MINIMUM_SECS);
-					let mut lightning_wallet_sync_interval = tokio::time::interval(
-						Duration::from_secs(lightning_wallet_sync_interval_secs),
-					);
-					lightning_wallet_sync_interval
-						.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
+		macro_rules! start_tx_based_sync_loop {
```

Comment:
It seems ldk devs aren't all equally enthusiastic about macros. Some even actively refactoring them out.

Can this be expressed with a normal fn?